### PR TITLE
Write the .gitpod.yml reference page.

### LINKED ISSUE #474: Documentation: Write "References - `.gitpod.yml`"
Relates to #461.

Use https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/data/gitpod-schema.json as the source of truth.

### PR DESCRIPTION
**Pull request by mikenikles**
Close #474.

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/604"></a>

### CODE REVIEW

File: src/routes/docs/references/gitpod-yml.md [hunk 1]
```diff
@@ -1,14 +1,312 @@
...
+gitConfig:
+  alias.st: status
+  core.autocrlf: input
+```
+
+Please refer to https://git-scm.com/docs/git-config#_values for a list of accepted values.
+
+| Type     | Default   |
+| -------- | --------- |
+| `object` | `<empty>` |
```
Comment (jankeromnes) [inline src/routes/docs/references/gitpod-yml.md:71]: 🤯 today I learned

File: src/routes/docs/references/gitpod-yml.md [hunk 2]
```diff
@@ -1,14 +1,312 @@
...
+
+Gitpod can add a comment with an “Open in Gitpod” button to your pull requests. Alternatively, you could [add a badge](#prebuildsaddbadge) to the pull request's description.
+
+| Type      | Default |
+| --------- | ------- |
+| `boolean` | `false` |
+
+### `prebuilds.addLabel`
+
+Deprecated.
```
Comment (jankeromnes) [inline src/routes/docs/references/gitpod-yml.md:107]: 😍 yes

File: src/routes/docs/references/gitpod-yml.md [hunk 3]
```diff
@@ -1,14 +1,312 @@
...
+
+### `tasks[n].env`
+
+Define environment variables that will be available in the workspace.
+
+Learn more about [Environment Variables](/docs/environment-variables) in the docs.
+
+| Type     | Default   |
+| -------- | --------- |
+| `object` | `<empty>` |
```
Comment (jankeromnes) [inline src/routes/docs/references/gitpod-yml.md:244]: 🤯 today I learned

File: src/routes/docs/references/gitpod-yml.md [hunk 4]
```diff
@@ -1,14 +1,312 @@
...
+
+Configure the VS Code editor.
+
+| Type     | Default   |
+| -------- | --------- |
+| `object` | `<empty>` |
+
+### `vscode.extensions`
+
+Define a list of extensions which should be installed for users of this workspace. The identifier of an extension is always `${publisher}.${name}`. For example: 'vscode.csharp'.
```
Comment (jankeromnes) [inline src/routes/docs/references/gitpod-yml.md:298]: Two thoughts:

- Should we mention that Gitpod sources from [OpenVSX](https://open-vsx.org)? And/or link to the Extensions docs?

- I think `vscode.csharp` doesn't exist, and `ms-dotnettools.csharp` has a [license](https://marketplace.visualstudio.com/items/ms-dotnettools.csharp/license) that seems incompatible with OpenVSX. Maybe use a different example? (E.g. I know `vscode.vim` exists.)

**Review by jankeromnes — APPROVED** [review]
Wow, that's awesome @mikenikles! 💯 🥇

Thanks a lot for adding this! 🙏 (Reminder: See also @ghuntley's comments in Slack.)

File: src/routes/docs/references/gitpod-yml.md (hunk 4 above)
Comment (mikenikles) [reply src/routes/docs/references/gitpod-yml.md:298]: Great points, done and done :).

**Comment by mikenikles** [comment]
Thanks @jankeromnes & @ghuntley for your feedback. It's all addressed in the latest preview deployment. Let me know if you have any other feedback 🙏.

**Comment by mikenikles** [comment]
Actually, let me merge this. Please open new issues if there's additional feedback.

**Pull request merged**